src/multisig/multisigTransactionsFactory.ts

```diff
@@ -0,0 +1,509 @@
+            functionName: options.functionName,
+            arguments: options.functionArguments,
+            abi: options.abi,
+        });
+        const dataParts = [
+            "proposeTransferExecute",
+            this.argSerializer.valuesToStrings([new AddressValue(options.to)])[0],
+            this.argSerializer.valuesToStrings([new BigUIntValue(options.egldAmount)])[0],
+            this.argSerializer.valuesToStrings([new OptionValue(new OptionType(new U64Type()), gasOption)])[0],
+            ...input.functionCall,
+        ];
+
+        return new TransactionBuilder({
+            config: this.config,
+            sender: sender,
+            receiver: options.multisigContract,
+            dataParts: dataParts,
+            gasLimit: options.gasLimit,
+            addDataMovementGas: true,
+        }).build();
+    }
+
+    /**
+     * Proposes a transaction that will transfer EGLD and/or execute a function
+     */
+    createTransactionForDeposit(sender: Address, options: resources.DepositExecuteInput): Transaction {
+        return this.createTransactionForExecute(sender, {
+            contract: options.multisigContract,
+            function: "deposit",
+            gasLimit: options.gasLimit ?? 0n,
+            arguments: [],
+            nativeTransferAmount: options.egldAmount,
+            tokenTransfers: options.tokenTransfers,
+        });
+    }
+
+    /**
+     * Proposes a transaction that will transfer ESDT tokens and/or execute a function
+     */
+    createTransactionForProposeTransferExecuteEsdt(
+        sender: Address,
+        options: resources.ProposeTransferExecuteEsdtInput,
+    ): Transaction {
+        const input = resources.ProposeTransferExecutInput.newFromTransferExecuteInput({
+            multisig: options.multisigContract,
+            to: options.to,
+            tokenTransfers: [],
+            functionName: options.functionName,
+            arguments: options.functionArguments,
+            abi: options.abi,
+        });
+        const tokenComputer = new TokenComputer();
+        const argsTyped = [];
+        for (const token of options.tokens) {
+            argsTyped.push({
+                token_identifier: new TokenIdentifierValue(
+                    tokenComputer.extractIdentifierFromExtendedIdentifier(token.token.identifier),
+                ),
+                token_nonce: new U64Value(token.token.nonce),
+                amount: new BigUIntValue(token.amount),
+            });
+        }
+        const dataParts = [
+            "proposeTransferExecuteEsdt",
+            ...this.argSerializer.valuesToStrings(
+                NativeSerializer.nativeToTypedValues(
+                    [options.to, argsTyped, options.gasLimit, VariadicValue.fromItems(...input.functionCall)],
+                    this.abi?.getEndpoint("proposeTransferExecuteEsdt") ??
+                        new EndpointDefinition("proposeTransferExecuteEsdt", [], [], new EndpointModifiers("", [])),
```

Comment:
Does this work if the Abi is not provided? 
Perhaps you could convert all values to typed values so the abi would not be needed.

Reply:
not, but I'll do this in a different pr